### PR TITLE
midipix platform support

### PR DESCRIPTION
**Pull request by Redfoxymoon**
## Description
This PR brings support for midipix, a new POSIX layer for windows, like cygwin.

### CODE REVIEW

File: library/entropy_poll.c [hunk 1]
```diff
@@ -95,7 +95,7 @@ int mbedtls_platform_entropy_poll( void *data, unsigned char *output, size_t len
  * Since there is no wrapper in the libc yet, use the generic syscall wrapper
  * available in GNU libc and compatible libc's (eg uClibc).
  */
-#if defined(__linux__) && defined(__GLIBC__)
+#if defined(__linux__) && defined(__GLIBC__) || defined(__midipix__)
```
Comment (danh-arm) [inline library/entropy_poll.c:98]: Better to not rely on operator precedence here; use parenthesis

**Comment by Redfoxymoon** [comment]
Is it better now?:-)

File: library/entropy_poll.c [hunk 2]
```diff
@@ -95,7 +95,7 @@ int mbedtls_platform_entropy_poll( void *data, unsigned char *output, size_t len
  * Since there is no wrapper in the libc yet, use the generic syscall wrapper
  * available in GNU libc and compatible libc's (eg uClibc).
  */
-#if defined(__linux__) && defined(__GLIBC__)
+#if (defined(__linux__) && defined(__GLIBC__) || defined(__midipix__))
```
Comment (danh-arm) [inline library/entropy_poll.c:98]: I was thinking more like `if ((defined(__linux__) && defined(__GLIBC__)) || defined(__midipix__))`

Comment (Redfoxymoon) [reply library/entropy_poll.c]: Sure:-)

**Comment by danh-arm** [comment]
1 failing test is an Mbed OS specific failure, which seems unrelated to this change.

**Comment by Redfoxymoon** [comment]
I can't see the details of the failed test either, something's up with the SSL setup(?)

**Comment by danh-arm** [comment]
The test results are only visible within Arm. In this case you don't need to worry about the failure.

**Comment by Redfoxymoon** [comment]
Oh I see. Nevermind me then:-)

**Pull request merged**